src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionXpathSingleFilterTest.java

```diff
@@ -400,6 +401,59 @@ public void testDecideById() throws Exception {
                 expected, removeSuppressed(expected, suppressed));
     }
 
+    /**
+     * This test is required to cover pitest mutation
+     * for reset of 'file' field in SuppressionXpathSingleFilter.
+     * This not possible to reproduce by natural execution of Checkstyle
+     * as config is never changed in runtime.
+     * Projects that use us by api can reproduce this case.
+     * We need to allow users to reset module property to default state.
+     *
+     * @throws Exception when there is problem to load Input file
+     */
+    @Test
+    public void testUpdateFilterSettingsInRunTime() throws Exception {
+        final File file = new File(getPath("InputSuppressionXpathSingleFilterComplexQuery.java"));
+
+        final SuppressionXpathSingleFilter filter = new SuppressionXpathSingleFilter();
+        filter.setChecks("MagicNumber");
+        filter.finishLocalSetup();
+
+        final Violation violation = new Violation(27, 21, TokenTypes.NUM_DOUBLE, "",
+                        "", null, null, null,
+                        MagicNumberCheck.class, null);
+
+        final FileContents fileContents =
+                new FileContents(new FileText(file, StandardCharsets.UTF_8.name()));
+
+        final TreeWalkerAuditEvent ev = new TreeWalkerAuditEvent(fileContents, file.getName(),
+                violation, JavaParser.parseFile(file, JavaParser.Options.WITHOUT_COMMENTS));
+
+        assertWithMessage("match ia expected as 'files' is defaulted")
+                .that(filter.accept(ev))
+                .isFalse();
+
+        // set non-default value for files
+        filter.setFiles(sanitizeWindowsPathForRegexp(file.getPath() + ".never.match"));
+        filter.finishLocalSetup();
+
+        assertWithMessage("no match is expected due to weird value in 'files'")
+                .that(filter.accept(ev))
+                .isTrue();
+
+        // reset files to default value of filter
+        filter.setFiles(null);
+        filter.finishLocalSetup();
+
+        assertWithMessage("match ia expected as 'files' is defaulted")
+                .that(filter.accept(ev))
+                .isFalse();
+    }
+
+    private static String sanitizeWindowsPathForRegexp(String value) {
+        return value.replaceAll("\\\\", "\\\\\\\\\\\\\\\\");
```

Comment:
https://github.com/checkstyle/checkstyle/blob/a89ddac62254f3b94a2d97231bf6a878563d74af/config/suppressions.xml#L12

Reply:
It was good form for humans to deal with, do you really think we need this fancy escaping for execution in windows in Junit?

Reply:
method is removed.